### PR TITLE
feat(core): enable prepending jvm flags from docker run

### LINKED ISSUE #3401: extend the docker-entry script to allow passing additional JVM flags
### Is your feature request related to a problem?

QuestDB [docker entry script](https://github.com/questdb/questdb/blob/7.1.3/core/docker-entrypoint.sh) keeps Java heap config default and relies on ergonomics. 

Users can pass arguments for the JVM, but this overwrites all default arguments. 
So I want to set e.g. `Xmx` then I also have to set the DATADIR and module path, logging config, etc. 

Example: `docker run questdb/questdb -Xmx5g -m io.questdb/io.questdb.ServerMain -d /var/lib/questdb`

### Describe the solution you'd like.

It would be nice to have a way to prepend additional JVM flags without having to explicitly include all the default arguments. So something like `docker run -e JVM_PREPEND="-Xmx5g" questdb/questdb` could work. 

### Describe alternatives you've considered.

_No response_

### Additional context.

This could be also useful for enabling the verbose GC log and other JVM toggles. 
QuestDB is very conservative with Java heap allocations, so most of the time it's not needed. but it would be still nice to have some convenience for situations when it is needed.

### PR DESCRIPTION
**Pull request by yifanstefanwang**
## Summary
Closes #3401 

## Testing Done
 1. Docker build locally which uses the updated script
 2. Docker run locally
 ```
➜  ~ docker run -e JVM_PREPEND="-Xmx5g" ac28b181eb77
Found on-demand JVM arguments: -Xmx5g, prepending to JVM args
No arguments found in the configuration, start with default arguments
Running as questdb user
Log configuration loaded from: /var/lib/questdb/conf/log.conf
2023-07-13T23:42:25.817836Z A server-main QuestDB 7.2.2-SNAPSHOT. Copyright (C) 2014-2023, all rights reserved.
2023-07-13T23:42:25.935165Z A server-main OS/Arch linux/arm64 [Vanilla,-1744679816, 64 bits, 4 processors]
 ```
3. Verified the java command ran in the contianer
```
➜  ~ docker ps
CONTAINER ID   IMAGE          COMMAND                  CREATED              STATUS              PORTS                          NAMES
9a2c7cf3ccae   ac28b181eb77   "/docker-entrypoint.…"   About a minute ago   Up About a minute   8812/tcp, 9000/tcp, 9009/tcp   beautiful_lamarr
```
```
# cd /proc
# cd 1
# cat cmdline
/app/bin/java-Xmx5g-XX:ErrorFile=/var/lib/questdb/db/hs_err_pid+%p.log-Dout=/var/lib/questdb/conf/log.conf-mio.questdb/io.questdb.ServerMain-d/var/lib/questdb-f# 
```

### CODE REVIEW

**Comment by puzpuzpuz** [comment]
@mariusgheorghies could you review this PR?

**Comment by yifanstefanwang** [comment]
> @mariusgheorghies could you review this PR?

Thanks @puzpuzpuz , do you have a group alternative reviewers in mind? still haven't got a review, and this branch started having churn of needing to rebase with master.

Thank you!

**Comment by puzpuzpuz** [comment]
> Thanks @puzpuzpuz , do you have a group alternative reviewers in mind? still haven't got a review yet on this.
> 
> Thank you!

Terribly sorry for the delay. I'm vacationing this week. Will do my best to review your PR on Monday.

File: core/docker-entrypoint.sh [hunk 1]
```diff
@@ -38,16 +38,23 @@ if [ "$IGNORE_DATA_ROOT_MOUNT_CHECK" = "false" ] && mount | grep "/root/.questdb
...
+    set -- $JAVA_COMMAND -XX:ErrorFile=${QUESTDB_DATA_DIR}/db/hs_err_pid+%p.log -Dout=${QUESTDB_DATA_DIR}/conf/log.conf -m io.questdb/io.questdb.ServerMain -d ${QUESTDB_DATA_DIR} -f
 else
     if [ "${1:0:1}" = '-' ]; then
         echo "Found config arguments $@"
-        set -- /app/bin/java "$@"
+        set -- $JAVA_COMMAND "$@"
     elif [ "$1" = "/app/bin/java" ]; then
         echo "Java binary arguments found, Non default arguments config run"
-        set -- "$@"
+        set -- $JVM_PREPEND "$@"
```
Comment (puzpuzpuz) [inline core/docker-entrypoint.sh:57]: Let's keep the old command here. The use case is `docker run <image-id> /app/bin/java the_remaining_args` - so, full control over the startup command.

If we use `set -- $JVM_PREPEND "$@"` here, command like `docker run -e JVM_PREPEND="-Xmx5g" 5caa247a98b1 /app/bin/java` leads to `gosu 10001:10001 -Xmx5g /app/bin/java` which has incorrect flag order.

File: core/docker-entrypoint.sh [hunk 2]
```diff
@@ -56,8 +63,10 @@ if [ "$(id -u)" = '0' ] && [ "${QUESTDB_DATA_DIR%/}" != "/root/.questdb" ] && [
     if [ "$DO_CHOWN" = "true" ]; then
         find_and_own_dir $QUESTDB_UID $QUESTDB_GID
     fi
+    echo "Executing: gosu $QUESTDB_UID:$QUESTDB_GID $@"
```
Comment (puzpuzpuz) [inline core/docker-entrypoint.sh:66]: nit: I don't know if we should print the full command - some users might have some sensitive arguments there. Same consideration for L71 (`echo "Executing: $@"`). WDYT?

Comment (yifanstefanwang) [reply core/docker-entrypoint.sh:66]: Gotcha, makes sense, will remove

File: core/docker-entrypoint.sh (hunk 1 above)
Comment (yifanstefanwang) [reply core/docker-entrypoint.sh:57]: Thanks for clarifying this use-case! Will fix

**Review by puzpuzpuz — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@yifanstefanwang thanks for the contribution!

**Pull request merged**